packages/sanity/src/datastores/cors/checkCors.ts

```diff
@@ -0,0 +1,21 @@
+import {SanityClient} from '@sanity/client'
+import {CorsOriginErrorOptions} from './CorsOriginError'
+
+export const checkCors = (client: SanityClient): Promise<CorsOriginErrorOptions> => {
+  const pingRequest = client.request({uri: '/ping', withCredentials: false}).then(() => true)
+  const userRequest = client
+    .request({uri: '/users/me', withCredentials: false})
+    .then(() => true)
+    .catch(() => false)
+
+  const promise = Promise.all([pingRequest, userRequest])
+    .then((res) => ({
+      isCorsError: res[0] && !res[1],
+      pingResponded: res[0],
+    }))
```

Comment:
I'd like to suggest using destructing here to improve readability :)
```suggestion
    .then(([pingResponded, userResponded]) => ({
      isCorsError: pingResponded && !userResponded,
      pingResponded,
    }))
```

Reply:
Agree, cleaner :)